WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java

```diff
@@ -363,32 +471,48 @@ private String getLanguageString(String languageCode, Locale displayLocale) {
             return "";
         }
 
-        Locale languageLocale = new Locale(localeInput(languageCode));
-        return languageLocale.getDisplayLanguage(displayLocale) + languageCode.substring(2);
+        Locale languageLocale = languageLocale(languageCode);
+        String displayLanguage = languageLocale.getDisplayLanguage(displayLocale);
+        String displayCountry = languageLocale.getDisplayCountry(displayLocale);
+
+        if (!TextUtils.isEmpty(displayCountry)) {
+            return displayLanguage + " (" + displayCountry + ")";
+        }
+        return displayLanguage;
     }
 
     /**
-     * Removes a {@link Preference} from the {@link PreferenceCategory} with the given key.
+     * Gets a preference and sets the {@link android.preference.Preference.OnPreferenceChangeListener}.
      */
-    private void removePreference(int categoryKey, Preference preference) {
-        if (preference == null) return;
+    private Preference getPref(int id) {
+        Preference pref = findPreference(getString(id));
+        pref.setOnPreferenceChangeListener(this);
+        return pref;
+    }
 
-        PreferenceCategory category = (PreferenceCategory) findPreference(getString(categoryKey));
+    /**
+     * Removes a {@link Preference} from the {@link PreferenceCategory} with the given key.
+     */
+    private void removePreference(int parentKey, int preference) {
+        PreferenceGroup parent = (PreferenceGroup) findPreference(getString(parentKey));
 
-        if (category != null) {
-            category.removePreference(preference);
+        if (parent != null) {
+            parent.removePreference(findPreference(getString(preference)));
         }
     }
 
-    private String localeInput(String languageCode) {
-        if (TextUtils.isEmpty(languageCode)) return "";
+    private Locale languageLocale(String languageCode) {
+        if (TextUtils.isEmpty(languageCode)) return Locale.getDefault();
 
-        return languageCode.substring(0, 2);
+        if (languageCode.length() > 2) {
+            return new Locale(languageCode.substring(0, 2), languageCode.substring(3));
+        }
+
+        return new Locale(languageCode);
     }
 
     private String firstLetterCapitalized(String input) {
         if (TextUtils.isEmpty(input)) return "";
-
         return input.substring(0, 1).toUpperCase() + input.substring(1);
```

Comment:
Same w/ 1 (even though it's a bit more obvious and it was here before you did this work)